### ISSUE TITLE
Support max height and max weight in routing for cargo trucks and RVs

### ISSUE THREAD

**Issue by biodranik** [organicmaps/organicmaps#1507]
There should be an easy UI to somehow input the current car parameters and take them into an account while building a route.
See https://wiki.openstreetmap.org/wiki/Key:maxweight and https://wiki.openstreetmap.org/wiki/Key:maxheight for more details.

**Comment by TheAdventurer64** [comment]
This is what Waze does:![image](https://user-images.githubusercontent.com/50059322/155056166-58da8d24-e4f4-4130-ad69-d780d03169ec.png)

**Comment by pastk** [comment]
I wonder what is the difference of these options?

**Comment by biodranik** [comment]
Yes, the main value is to understand, what exactly should be tuned in the algorithm, and which restrictions should be analyzed/taken from the OSM data.

**Comment by SirTypesALot** [comment]
Could we also add routing support for mopeds?

**Comment by biodranik** [comment]
Everything is possible if you can list all differences from car navigation. If we have such a list, then it is possible to develop a proper algorithm. From my understanding, one is that there are some roads where only mopeds are allowed. What else?

**Comment by SirTypesALot** [comment]
Yes, that is correct. In the Netherlands, mopeds are sometime permitted to use cycling paths, and sometimes not. Do note that there are different classifications of mopeds in the Netherlands. I'll follow up soon when I can give you a clear answer about the classification.

**Comment by gtrus21** [comment]
Есть ещё ограничения для автобусов. А автобус это или нет - зависит от количества мест в салоне

**Comment by gtrus21** [comment]
Есть параметр нагрузки на ось (причем на каждую отдельную в зависимости от количества осей). Допустимая общая нагрузка на авто
![R0zx-3w7TH0](https://user-images.githubusercontent.com/69676792/205248534-eaea27a9-aef3-4de3-84f3-79da14ee3568.jpg)
![f0d095595d40e09e117da8cf3073bfb8](https://user-images.githubusercontent.com/69676792/205248541-7a273cbf-f15c-435a-997b-7224d49ded0f.jpeg)

**Comment by gtrus21** [comment]
Запрет обгона для грузовиков. Длина грузовика
![znaki-dlia-gruzovykh-avtomobilei-1 (1)_1](https://user-images.githubusercontent.com/69676792/205249276-a830084b-04ca-48e5-86de-15164e4ab8d6.jpg)

**Comment by gtrus21** [comment]
> Everything is possible if you can list all differences from car navigation. If we have such a list, then it is possible to develop a proper algorithm. From my understanding, one is that there are some roads where only mopeds are allowed. What else?

Мопеды должны двигаться только по правой полосе, значит в зависимости от количества полос им может быть запрещен поворот налево

**Comment by gtrus21** [comment]
То есть нужна возможность создавать предметы транспортных средств (машина 1, машина 2) и прицепов. Если у человека две машины (на работу едет на Ладе, а на работе водит грузовик), то переключается между машинами в один клик.
У каждой машины нужна возможность указывать ее тип (обязательно, ведь это может быть как легковушка на 4 тонны, так и грузовик на 3 тонны), количество осей, вес транспорта, длину.
К каждой машине нужна возможность добавлять заготовленные (заранее созданные пользователем) профили прицепов в любом количестве (не забывайте про автопоезда с 5 прицепами).
Ну а уже в быстром доступе нужна кнопка добавления веса груза, который добавится к весу машины

**Comment by gtrus21** [comment]
Ну и нужна галочка, что данное средство является общественным или такси, чтобы ему разрешалось движение по выделенной полосе

**Comment by DavyLandman** [comment]
For driving with a car that has a lower limit than the roads allow, for example an legal limit for a trailer, or a physical limit of an older camper for example, it would be great to say: hey, I can only drive 90km/h max, so please take that into account with the estimated total time.

Not many apps have this, and it would be so nice to have a better estimate. Especially if your for example driving on roads in Germany, that allow for 130km/h, but you have a trailer that restricts you to 80km/h


I was looking at graphhopper and noticed they provide you a way to say: hey this is my custom speed limit:

![image](https://github.com/organicmaps/organicmaps/assets/179603/e0e41187-a6b4-4c68-8e9e-f60c30274bc1)

**Comment by MBj1703** [comment]
Any progress on this?
How can I help?

**Comment by pastk** [comment]
An elaborated UI/UX design is needed first.

Then if you're a developer you can start implementing it 🙏

**Comment by MBj1703** [comment]
OsmAnd is also able: https://osmand.net

**Comment by Lin-Admin** [comment]
Has any expert an estimate how much time it would take to implement this fantastic idea?

**Comment by Lin-Admin** [comment]
@MBj1703: > OsmAnd is also able: https://osmand.net 

I would not mind you to advertise for OsmAnd if they would not charge for the cards :-(

**Comment by tverlaan** [comment]
OsmAnd does implement most of it, but not the maximum speed of the vehicle (at least on iOS). I came here because of this. I drive an older RV so I don't drive max speeds at all. Taking into account emission zones would also be very helpful.

**Comment by rbsns** [comment]
This would be great when driving high vehicles, as some returns and underpasses have height restrictions.

**Comment by LasDivine** [comment]
Hey

Glad to see a discussion on big vehicles

There are a dozen CMV apps out there for trucker specifically. They deal with navigation that is specifically designed to accommodate tractor trailer condos that have a lot of common sense restrictions.

You could look at hammer, trucker path, or truck map for examples. 

As of now, there are no truck map apps that are as good as let's say a Garmin for instance. A Garmin that has a lot of CMV customization features and with the ability to ensure you don't take any roads that are not actually accessible to Giant trucks and trailers. 

I'd really like to work on that as a driver specifically. I'm not a SWE so please don't rely on me on that end but I have been driving for nearly 4 years so I believe I have a great idea of how to build up the features that are necessary for a commercial motor vehicle driver AKA truck driver

Quick search along route:
        Rest areas, 
        Truck parking
        Weight station
Fuel station
        Specific to CMV
        Distill #2 gas stations with DEF
        Filter
Truck profile:
        Truck dimension
        Trailer dimension 
Alerts 
        Accidents, upcoming traffic
        Weight station (open/closed) 
User submission (+)
        Police
        Obstructions
        Weight station status
        Accidents
        Fuel prices 
Route planning
        Weight restriction
        Height restrictions
        Length restrictions
        Material
        Toll roads
Interface UI
        Zoom +-
        HUB arrows 
        Quick toggle mute

![Image](https://github.com/user-attachments/assets/f24a6a65-e0d4-4984-91b7-e0f880f88de7)
